specs/waku/v2/waku-store.md

```diff
@@ -2,36 +2,94 @@
 title: Waku
 version: 2.0.0-beta1
 status: Draft
-authors: Oskar Thorén <oskar@status.im>, Dean Eigenmann <dean@status.im>
+authors: Oskar Thorén <oskar@status.im>, Dean Eigenmann <dean@status.im>, Sanaz Taheri <sanaz@status.im>
 ---
 
 # Table of Contents
 
+- [Table of Contents](#table-of-contents)
 - [Abstract](#abstract)
+  - [Waku Store Protocol](#waku-store-protocol)
+  - [Pagination](#pagination)
 - [Wire Specification](#wire-specification)
-  * [Protobuf](#protobuf)
+  - [Protobuf](#protobuf)
+    - [Index](#index)
+    - [PagingInfo](#paginginfo)
+    - [HistoryQuery](#historyquery)
+    - [HistoryResponse](#historyresponse)
+    - [HistoryRPC](#historyrpc)
 - [Changelog](#changelog)
 - [Copyright](#copyright)
 
 # Abstract
 
-`WakuStore` is a protocol to enable querying of messages received through relay protocol and stored by other nodes.
+`WakuStore` is a protocol to enable querying of messages received through relay protocol and stored by other nodes. It also support pagination for more efficient querying of historical messages.
+
+## Pagination
+
+Nodes supporting the store protocol MAY support pagination. When a node supports pagination, a requester node can set the number of `WakuMessage`s to return and the offset to start from. The queried node MUST then use this information to return messages in the queried range. The details of the pagination are explained next.
+
+To perform pagination, each `WakuMessage` stored at a Waku node running the store protocol is associated with a unique `Index`. Once a `WakuMessage` is received by a node, it will generate the `Index` of the  `WakuMessage` and then saves the (`Index`, `WakuMessage`) together in its local storage. An `Index` encapsulates fields useful for paging, these fields are a `digest` and a `receivedTime`. The `receivedTime` indicates the time at which the `WakuMessage` is received by the store node and the `digest` is the hash of the received `WakuMessage`.  
```

Comment:
So I think that the `Index` is an implementation detail and therefore not required to be in the spec, wdyt @oskarth?

Reply:
Index is a protobuf field so it should be explained to same extent. Agree it can be more minimal though (see duplication comment)